nav2_map_server/src/map_server/map_server.cpp

```diff
@@ -206,6 +217,7 @@ bool MapServer::loadMapResponseFromYaml(
   const std::string & yaml_file,
   std::shared_ptr<nav2_msgs::srv::LoadMap::Response> response)
 {
+  grid_available_ = false;
```

Comment:
This seems redundant, and in some cases could be wrong. If we have a map already and a request comes in to load another that is invalid, there is still that initial map available. I think this should still be `true` because there is still indeed something available for use -- though the most recent request was not valid

Reply:
This is subjective though - I could see the argument for the opposite to be true

Reply:
I removed it. It's not obvious what to do here. If the robot just used an elevator, it would be better not to use the outdated map, if you only reached the border of a map before switching to the next one, you could still use it.